### PR TITLE
[297]: Links + embed

### PR DESCRIPTION
**Pull request by philipyoungg**
CleanShot 2021-03-24 at 20 57 21@2x
CleanShot 2021-03-24 at 20 56 43@2x
CleanShot 2021-03-24 at 20 56 19@2x

- Update share component to support full text (instead of prefixed with URL)
- Replace all naked string to url builder

### CODE REVIEW

File: src/components/collection/ViewCollection.tsx [hunk 1]
```diff
@@ -302,198 +367,272 @@ class ViewCollection extends React.Component<Props, State> {
...
               <Col>
                 <div className="flex items-center justify-between">
                   <h1>{title}</h1>
-                  {!!id &&
+                  {!!id && (
                     <h3 style={{ marginLeft: "1rem" }}>
-                      <Share suffix={`collection/${id}`} />
+                      <Share
+                        variant={"prefixedWithHostname"}
+                        text={collectionURL(id)}
```
Comment (philipyoungg) [inline src/components/collection/ViewCollection.tsx:431]: Only this part has changed in this file

File: src/components/item/ViewItem.tsx [hunk 2]
```diff
@@ -144,184 +154,316 @@ class ViewItem extends React.Component<Props, State> {
...
+                    )}
+                    <h3>
+                      <Share
+                        variant="fullText"
+                        iconComponent={<FaFileCode />}
+                        text={iframeItemEmbedCodeURL(
+                          this.props.item.id,
+                          this.props.item.title || ""
+                        )}
+                      />
```
Comment (philipyoungg) [inline src/components/item/ViewItem.tsx:269]: Only this part has changed in this file

File: src/components/metadata/CollectionEditor.tsx [hunk 3]
```diff
@@ -1419,75 +1947,165 @@ class CollectionEditorClass extends React.Component<Props, State> {
...
+                  {!!this.state.originalCollection.id && (
+                    <h3 style={{ marginLeft: "1rem" }}>
+                      <Share
+                        color="dark-gray"
+                        variant="prefixedWithHostname"
+                        text={collectionURL(this.state.originalCollection.id)}
+                      />
+                    </h3>
+                  )}
+                </div>
```
Comment (philipyoungg) [inline src/components/metadata/CollectionEditor.tsx:1998]: Only this part has changed in this file

File: src/components/metadata/ItemEditor.tsx [hunk 4]
```diff
@@ -2356,415 +3277,978 @@ class ItemEditorClass extends React.Component<Props, State> {
...
+                        variant="fullText"
+                        iconComponent={<FaFileCode />}
+                        text={iframeItemEmbedCodeURL(
+                          this.props.item.id,
+                          this.props.item.title || ""
+                        )}
+                      />
+                    </h3>
+                  </div>
+                )}
```
Comment (philipyoungg) [inline src/components/metadata/ItemEditor.tsx:3496]: Only this part has changed in this file

**Review by matthewberryman — APPROVED** [review]

**Pull request merged**